rspace/src/main/scala/coop/rchain/rspace/pure/PureRSpace.scala

```diff
@@ -5,26 +5,32 @@ import coop.rchain.rspace._
 
 import scala.collection.immutable.Seq
 
-class PureRSpace[F[_], C, P, E, A, R, K](space: FreudianSpace[C, P, E, A, R, K]) {
+class PureRSpace[F[_]: Sync, C, P, E, A, R, K](space: FreudianSpace[C, P, E, A, R, K]) {
 
   def consume(channels: Seq[C], patterns: Seq[P], continuation: K, persist: Boolean)(
-      implicit m: Match[P, E, A, R],
-      s: Sync[F]): F[Either[E, Option[(K, Seq[R])]]] =
-    s.delay(space.consume(channels, patterns, continuation, persist))
+      implicit m: Match[P, E, A, R]): F[Either[E, Option[(K, Seq[R])]]] =
+    Sync[F].delay(space.consume(channels, patterns, continuation, persist))
 
   def install(channels: Seq[C], patterns: Seq[P], continuation: K)(
-      implicit m: Match[P, E, A, R],
-      s: Sync[F]): F[Option[(K, Seq[R])]] =
-    s.delay(space.install(channels, patterns, continuation))
+      implicit m: Match[P, E, A, R]): F[Option[(K, Seq[R])]] =
+    Sync[F].delay(space.install(channels, patterns, continuation))
 
   def produce(channel: C, data: A, persist: Boolean)(
-      implicit m: Match[P, E, A, R],
-      s: Sync[F]): F[Either[E, Option[(K, Seq[R])]]] =
-    s.delay(space.produce(channel, data, persist))
+      implicit m: Match[P, E, A, R]): F[Either[E, Option[(K, Seq[R])]]] =
+    Sync[F].delay(space.produce(channel, data, persist))
 
-  def createCheckpoint()(s: Sync[F]): F[Checkpoint] = s.delay(space.createCheckpoint())
+  def createCheckpoint(): F[Checkpoint] = Sync[F].delay(space.createCheckpoint())
 
-  def reset(hash: Blake2b256Hash)(s: Sync[F]): F[Unit] = s.delay(space.reset(hash))
+  def reset(hash: Blake2b256Hash): F[Unit] = Sync[F].delay(space.reset(hash))
 
-  def close()(s: Sync[F]): F[Unit] = s.delay(space.close())
+  def close(): F[Unit] = Sync[F].delay(space.close())
+}
+
+object PureRSpace {
+  def apply[F[_]](implicit F: Sync[F]): PureRSpaceApplyBuilders[F] = new PureRSpaceApplyBuilders(F)
+
+  final class PureRSpaceApplyBuilders[F[_]](val F: Sync[F]) extends AnyVal {
+    def of[C, P, E, A, R, K](space: FreudianSpace[C, P, E, A, R, K]) =
```

Comment:
Is something preventing you from adding this method directly to the enclosing `object`? There would be less ceremony